### PR TITLE
enable -flto for unix systems

### PR DESCRIPTION
**Pull request by nmanthey**
These changes might improve the performance of CBMC, as well as allow to spot compilation problems earlier.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [x] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by nmanthey** [comment]
@peterschrammel: I wonder whether -flto is useful to be included for the binary that'll be submitted to SVCOMP. If you have resources for another test, that would be helpful.

File: src/common [hunk 1]
```diff
@@ -35,6 +35,9 @@ endif
   CFLAGS ?= -Wall -O2
   CP_CFLAGS = -MMD -MP
   CXXFLAGS ?= -Wall -O2
+  CXXFLAGS += -flto
+  LINKFLAGS += -flto
+  AR = gcc-ar
```
Comment (chrisr-diffblue) [inline src/common:40]: I guess CI will probably tell us this, but will this work with clang builds?

Comment (peterschrammel) [reply src/common:40]: Is this feature supposed to be stable in g++-5 ?

Comment (nmanthey) [reply src/common:40]: afaict, yes.

**Comment by nmanthey** [comment]
I wonder whether the bug in CI is an flto stability error, or an actual problem in the code base that's now uncovered.

**Comment by TGWDB** [comment]
Closing as (as discussed in #6011 ) this may not be stable/suitable.

**Pull request closed without merging**